src/contradictory_claims/data/preprocess_cord.py

```diff
@@ -27,7 +34,147 @@ def filter_metadata_for_covid19(metadata_path: str, virus_lex_path: str, pub_dat
         covid_19_terms = f.read().splitlines()
         covid_19_term_pattern = '|'.join([i.lower() for i in covid_19_terms])
 
-    covid19_df = metadata_df.loc[metadata_df.title_abstract.str.contains(covid_19_term_pattern)]
-    covid19_df = covid19_df.loc[metadata_df['publish_time'] > pub_date_cutoff]
+    covid19_df = metadata_df.loc[metadata_df.title_abstract.str.contains(covid_19_term_pattern)]\
+                            .copy().reset_index(drop=True)
+
+    covid19_df['publish_time'] = pd.to_datetime(covid19_df['publish_time'], format='%d-%m-%Y')
+    covid19_df = covid19_df.loc[covid19_df['publish_time'] > datetime.strptime(pub_date_cutoff, "%Y-%m-%d")]\
+                           .copy().reset_index(drop=True)
 
     return covid19_df
+
+
+def extract_json_to_dataframe(covid19_metadata: pd.DataFrame,
+                              json_text_file_dir: str,
+                              json_temp_path: str,
+                              pdf_filenames: List[str],
+                              pmc_filenames: List[str]):
+    """
+    Extract publications text from json files for a specified set of filenames and store in a dataframe.
+
+    :param covid19_metadata: pandas dataframe, output of filter_metadata_for_covid19()
+    :param json_text_file_dir: path to zip directory containing json files
+    :param json_temp_path: path for temporary file storage
+    :param pdf_filenames: list of pdf file names to extract
+    :param pmc_filenames: list of pmc file names to extract
+    :return: Dataframe of publication texts for the specified filenames
+    """
+    covid19_dict = {}
+
+    # Replace characters with their readable format
+    replace_dict = {'â€œ': '“',
+                    'â€': '”',
+                    'â€™': '’',
+                    'â€˜': '‘',
+                    'â€”': '–',
+                    'â€“': '—',
+                    'â€¢': '-',
+                    'â€¦': '…'}
+
+    # TODO: Parallelize the code below
+    with ZipFile(json_text_file_dir, 'r') as zipobj:
+        list_of_filenames = zipobj.namelist()
+        # print('Number of files to iterate over:',len(list_of_filenames))
+        k = 0
+        # iter_num = 0
+        for filename in list_of_filenames:
+            # iter_num = iter_num + 1
+            # Check filename ends with json and file exists in filtered list of cord papers
+            if (filename in pdf_filenames) or (filename in pmc_filenames):
+                zipobj.extract(filename, json_temp_path)
+                json_op = []
```

Comment:
what's json_op doing? Name isn't descriptive

Reply:
Ah good catch. It just stores the json data from the file. I'll rename it to json_output...